src/Pages/AddItem.js

```diff
@@ -1,19 +1,26 @@
 import React, { useState } from 'react';
+import { useCollection } from 'react-firebase-hooks/firestore';
 import { db } from '../lib/firebase';
 
 export default function AddItem({ token }) {
+  const [list, loading, error] = useCollection(db.collection(token));
   const [item, setItem] = useState('');
   const [urgency, setUrgency] = useState(7);
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    db.collection(token).add({
-      item,
-      urgency,
-      lastPurchased: null,
-    });
-    setUrgency(7);
-    setItem('');
+
+    if (validateNewListItem(item, list)) {
+      db.collection(token).add({
+        item,
+        urgency,
+        lastPurchased: null,
+      });
+      setUrgency(7);
+      setItem('');
+    } else {
+      console.log('Item already exists');
```

Comment:
Looks like we're missing a small piece of our AC. We just need to handle the error message 🤠  There's a great example of my preferred way of doing it, updating our UI to show a helpful message near the form, [here](https://github.com/the-collab-lab/tcl-33-smart-shopping-list/blob/main/src/components/JoinList.js)

Reply:
Thanks for pointing this out. It's now updated. I'd love feedback on the implementation if possible.

Reply:
Looks great!